packages/contracts/src/base/CredbullBaseVault.sol

```diff
@@ -53,12 +53,12 @@ abstract contract CredbullBaseVault is ICredbull, ERC4626, Pausable {
      *
      * @param baseVaultParams - Base vault parameters
      */
-    constructor(BaseVaultParams memory baseVaultParams)
+    constructor(ICredbull.BaseVaultParams memory baseVaultParams)
```

Comment:
I don't think this is the fix Ian had in mind, @Krishnakumarskr 
He was much more about splitting the `VaultParams` into its components and having those in the appropriate locations. It is not simply about ease of invocation. It is a Software Principle after all (Modularity). 
So, the fact that it touches lots of locations in the codebase is a given!

This change is simply changing how the 'params' are referenced. It is also counter to the use of an `interface` to encapsulate the params. If you use an interface, then you want to realise it, not simply directly reference it.